src/VirtualClient/VirtualClient.Actions/MLPerfTraining/MLPerfTrainingExecutor.cs

```diff
@@ -0,0 +1,515 @@
+                // add user in docker group
+                await this.ExecuteCommandAsync("usermod", $"-aG docker {this.Username}", this.ExecutionDirectoryPath, cancellationToken);
+
+                // Setup Environment
+                await this.SetupEnvironmentAsync(cancellationToken);
+                state.Initialized = true;
+                await this.stateManager.SaveStateAsync<MLPerfState>($"{nameof(MLPerfState)}", state, cancellationToken);
+            }
+        }
+
+        /// <summary>
+        /// Creates setup for MLPerf Training workload.
+        /// </summary>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        protected async Task SetupEnvironmentAsync(CancellationToken cancellationToken)
+        {
+            string dockerImageCommand = $"docker build --pull -t {this.GetContainerName()} .";
+            string dockerRunCommand = $"docker run --runtime=nvidia {this.GetContainerName()}";
+
+            await this.ExecuteCommandAsync(
+                "sudo",
+                dockerImageCommand,
+                this.ExecutionDirectoryPath,
+                cancellationToken);
+
+            await this.ExecuteCommandAsync(
+                "sudo",
+                dockerRunCommand,
+                this.ExecutionDirectoryPath,
+                cancellationToken);
+        }
+
+        /// <summary>
+        /// Gets the container name created by MLPerf Training.
+        /// </summary>
+        /// <returns>Container name created by MLPerf Training</returns>
+        /// <exception cref="WorkloadException"></exception>
+        protected string GetContainerName()
+        {
+            // Update this function to accomodate other architectures
+            if (this.Platform == PlatformID.Unix && this.CpuArchitecture == Architecture.X64)
+            {
+                return $"mlperf-training-{this.Username}-x86_64:{this.ContainerName}";
+            }
+            else if (this.Platform == PlatformID.Unix && this.CpuArchitecture == Architecture.Arm64)
+            {
+                return $"mlperf-training-{this.Username}-arm64:{this.ContainerName}";
+            }
+            else
+            {
+                throw new WorkloadException(
+                    $"The container name is not defined for the current platform/architecture " +
+                    $"{PlatformSpecifics.GetPlatformArchitectureName(this.Platform, this.CpuArchitecture)}.",
+                    ErrorReason.PlatformNotSupported);
+            }
+        }
+
+        /// <summary>
+        /// Executes the MLPerf Training workload.
+        /// </summary>
+        protected override async Task ExecuteAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            this.PrepareBenchmarkConfigsAndScenarios();
+            string shardsPath = this.PlatformSpecifics.Combine("/mlperftraining0", $"{this.DataPath}", "mlperf-training-package", "hdf5", "training-4320");
```

Comment:
Please add documents on how we are preparing the data and the expected data structure

Reply:
Sure, I will document the procedure for preparing the data and the model checkpoints for training